### DISCUSSION TITLE
How to calculate proper motions

### DISCUSSION THREAD

**Discussion by carbaior** [skyfielders/python-skyfield#923] [Q&A] [answered]
Hello All,

I'm trying to use Skyfield to compute proper motion speeds, but the values obtained are sometimes correct and sometimes very incorrect.

Please take a look at this code:

```
#!/usr/bin/python3

from skyfield.api import Star, load
from skyfield.data import hipparcos

def ecliptic_latlon_icrs(hip,t):
	S = Star.from_dataframe(df.loc[hip])
	tt = ts.utc(t,1,1)
	apparent=earth.at(tt).observe(S).apparent()
	lat, lon, distance = apparent.ecliptic_latlon()
	return lat.degrees, lon.degrees

def radec_arc_speed(hip):
	S = Star.from_dataframe(df.loc[hip])
	vdec=S.dec_mas_per_year
	vra=S.ra_mas_per_year
	v=(vdec*vdec+vra*vra)**0.5
	return round(v)

def latlon_arc_speed(hip):
	lat1,lon1=ecliptic_latlon_icrs(hip,2050)
	lat2,lon2=ecliptic_latlon_icrs(hip,1950)
	vlat=(lat1-lat2)
	vlon=(lon1-lon2)
	v=(vlat*vlat+vlon*vlon)**0.5
	#degrees/century to minutes to seconds to millis/year:
	v=v*60*60*1000/100
	return round(v)
	
with load.open(hipparcos.URL) as f:
    df = hipparcos.load_dataframe(f)

planets = load('de421.bsp')
earth = planets['earth']

ts = load.timescale()

print ("Star\t\tArc_Speed_radec\tArc_Speed_latlon")
for star in [746,17378,12777,86974,89937,96100]:
	radec_speed  = radec_arc_speed(star)
	latlon_speed = latlon_arc_speed(star)
	print (f"   HIP_{star}\t{radec_speed}\t\t{latlon_speed}")
```
	
This produces the following output:
```

Star		Arc_Speed_radec	Arc_Speed_latlon
   HIP_746	        554	        664
   HIP_17378	        748	        751
   HIP_12777	        346		389
   HIP_86974	        805	        867
   HIP_89937	        637	       5500
   HIP_96100	       1839	      11464
```

Why do I get such different values for HIP_89937 and HIP_96100?

**Comment by brandon-rhodes** [comment]
> Why do I get such different values for HIP_89937 and HIP_96100?

A first guess would be that those two stars have faster proper motions across the sky. Maybe check their proper motion in the catalog and see if it's higher than for the other stars?

Maybe you could provide more context about why you expect these six stars to have similar proper motions. Are they part of a star cluster that should be moving across the sky with fairly similar speed?

**Reply by carbaior** [reply]
Hi Brandon,

Those stars are just a small sample of a bigger set I'm working on, they are not related in any way.

I posted them to illustrate the problem I have. My current understanding is that the arc speed of a star has to be the same regardless of the system of coordinates used (its magnitude being the hypotenuse of its orthogonal components in any case).

So the arc_speed_radec (obtained from dec_mas_per_year and ra_mas_per_year properties) should be the same or very similar to that obtained from lat_mas_per_year and lon_mas_per_year (computed by lanlon_arc_speed function after comparing positions).

For the first 4 stars, this is the case. But for the last two, discrepancy is very high. Why?

**Reply by brandon-rhodes** [reply, marked as answer]
Oh! I thought you were asking about the discrepancy between rows, but now I see you are asking about the columns.

The problem is likely that you are treating polar coordinates like rectangular coordinates. Because lines of latitude get closer together near the pole, a degree of latitude means less and less actual motion. The formulae, I believe, are here:

https://en.wikipedia.org/wiki/Proper_motion

**Reply by carbaior** [reply]
¡That's correct, thanks a lot!

The correct formula for radec_v would be then: ` v=(vdec**2+(vra**2)*(math.cos(<star.dec>)**2))**0.5`

I was computing the arc_speed in both systems as a way to validate the results of ` latlon_arc_speed()`.
But as this magnitude is dependent on the distance from their respective poles, which are different, I can forget about trying this validation.

As I will work mainly with latitudinal speed, which is not dependent on the star's latitude, I think I can simply use `vlat = (lat1-lat2)` being lat1 and lat2 the latitudes of the same star after a 1 year period as a valid estimate.

Thanks again,
Carlos
